src/main/java/io/druid/data/input/MapBasedRow.java

```diff
@@ -126,7 +126,8 @@ public long getLongMetric(String metric)
       return ((Number) metricValue).longValue();
     } else if (metricValue instanceof String) {
       try {
-        return Long.valueOf(((String) metricValue).replace(",", ""));
+        String s = ((String) metricValue).replace(",", "");
+        return s.contains(".") ? Double.valueOf(s).longValue() : Long.valueOf(s);
```

Comment:
added, 
btw Long.valueOf("1L") or Long.valueOf("1l") is not permitted as it says in the doc.